llvm/include/llvm/Analysis/IVDescriptors.h

```diff
@@ -35,6 +35,7 @@ enum class RecurKind {
   // clang-format off
   None,     ///< Not a recurrence.
   Add,      ///< Sum of integers.
+  Sub,      ///< Subtraction of integers
```

Comment:
You will need to update a number of places with switches over recurrence kind, in SLPVectorizer and possibly others

Probably also should have a SLP test to make sure its handled correctly.